### ISSUE TITLE
flaw in packetParser.cpp 'readPacket' rejects valid packets

### ISSUE THREAD

**Issue by ScottFerg56** [adafruit/Adafruit_BluefruitLE_nRF51#23]
This file and function appear in three examples that I could find.

The 'readPacket' function resets the packet when a '!' byte is seen:
     if (c == '!') { replyidx = 0; }
    (line 101 after ble->read)
Unfortunately, '!' can be a valid value for a parameter or checksum byte in the packet. This flaw causes certain valid packets to be rejected with a "Checksum mismatch". For example, a color picker packet with any of the RGB values set to 33 ('!') will be rejected. Likewise a color picker packet whose RGB values sum to 122 will generate a checksum value of '!' and be rejected.

**Comment by petermichaux** [comment]
I agree this is confusing. It could very well be that a byte contains the value 33 as a color.

**Comment by petermichaux** [comment]
@hathach, any insights you can share on this?

**Comment by ladyada** [comment]
hathach didnt write it, i did :)
